failed to pull image by proxy-cache after change /etc/hosts

build 3833
1. docker pull busybox by proxy cache
2. change /etc/hosts of harbor-core
![image](https://user-images.githubusercontent.com/188115/90959521-3b72da00-e4ce-11ea-8a45-cf06873bf052.png)
3. docker pull busybox by proxy cache again, it will fail with 500 Internal Server Error
![image](https://user-images.githubusercontent.com/188115/90959545-55acb800-e4ce-11ea-9808-b4d5f303d725.png)
4. log in harbor-core show timeout error
![image](https://user-images.githubusercontent.com/188115/90959562-6e1cd280-e4ce-11ea-8daa-f41ab52fb51f.png)

This does not meet the description of test case

I got a similar issue like this. But what I did is not to change the hosts but change the credentials of the docker hub after the image cached successfully in the harbor. 

This is the core log after changed the credentials of the docker hub and pull the cached image again.

```
[ERROR] [/lib/http/error.go:54]: {"errors":[{"code":"UNKNOWN","message":"unknown: login to dockerhub error: login to dockerhub error: {\"detail\":\"Incorrect authentication credentials\"}\n\n"}]}
```

And this is the log of docker pull command
```
$ docker pull harbor.10.92.233.6.xip.io/proxy/library/hello-world:latest
Error response from daemon: received unexpected HTTP status: 500 Internal Server Error
```

--- a/src/server/middleware/repoproxy/proxy.go
+++ b/src/server/middleware/repoproxy/proxy.go
@@ -111,7 +111,19 @@ func handleManifest(w http.ResponseWriter, r *http.Request, next http.Handler) e
 		return nil
 	}
 	log.Debugf("the tag is %v, digest is %v", art.Tag, art.Digest)
-	man, err := proxyCtl.ProxyManifest(ctx, p, art)
+	err = proxyManifest(ctx, w, r, next, proxyCtl, p, art)
+	if err != nil {
+		if errors.IsNotFoundErr(err) {
+			return err
+		}
+		log.Warningf("Proxy to remote failed, fallback to local repo, error: %v", err)
+		next.ServeHTTP(w, r)
+	}
+	return nil
+}
+
+func proxyManifest(ctx context.Context, w http.ResponseWriter, r *http.Request, next http.Handler, ctl proxy.Controller, p *models.Project, art lib.ArtifactInfo) error {
+	man, err := ctl.ProxyManifest(ctx, p, art)
 	if err != nil {
 		return err
 	}

